### ISSUE TITLE
driver.reset()

### ISSUE THREAD

**Issue by Scisaga** [MachinePublishers/jBrowserDriver#204]
when call driver.reset() in crawler threads, eg. change a upstreaming proxy, remote JBrowserDriverServer act unreasonable. both thread num and mem consume grow rapidly.

related code:
```
Context.java Line 90 - 99
  void reset(JBrowserDriverServer driver) {
    removeItems();
    synchronized (lock) {
      setUpUserDataDir();
      ContextItem newContext = new ContextItem();
      newContext.init(driver, this);
      items.add(newContext);
      itemMap.put(newContext.itemId.get(), newContext);
    }
  }
```

since removeItems() just hide web elements using Toolkit util according to your code, may be the related threads and mem is not released. i don't quit understand this part of your code, may be you can give me some hints, thanks.

BTW. is there any change proxy method for driver?

**Comment by hollingsworthd** [comment]
Thanks for the info. You're right, [Stage.close()](https://docs.oracle.com/javase/8/javafx/api/javafx/stage/Stage.html#close--) doesn't actually free resources which is a bug.

The goal of the `reset` API is to re-initialize the state of the browser without actually re-initializing or classloading JavaFX or starting a new Java child process. Calling `new JBrowserDriver()` is expensive--although in context of loading a typical modern webpage (which is very expensive) it's not too bad.

`Context` is equivalent to the state of a browser. `ContextItem`s are windows within that browser, that share the overall browser state, and each `JBrowserDriver` is a wholly separate browser instance running in a separate process.

Basically `reset` closes each browser window, gives you a new instance of `WebView` which is the JavaFX browser built on WebKit, clears cookies, user data, and other settings.

**Comment by hollingsworthd** [comment]
> BTW. is there any change proxy method for driver?

Not aside from `reset`

**Comment by andreaskas** [comment]
Anu updates on this issue? Memory leaks using reset() are still here. Thanks

**Comment by hollingsworthd** [comment]
None yet

**Comment by Gintasz** [comment]
Any updates?
